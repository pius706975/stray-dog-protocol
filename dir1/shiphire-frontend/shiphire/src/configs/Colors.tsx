export const Color = {
    bgColor: "#f0f6fb",
    softSecBgPrimary: "#bcd3e6",
    softGreyBgPrimary: "#cbcdd1",
    primaryColor: "#216dab",
    softPrimaryColor: "#6499c4",
    secColor: "#9cd2ff",
    softSecColor: "#d7edff",
    darkTextColor: "#4d5963",
    lightTextColor: "#f0f6fb",
    bgSuccessColor: "#e9faec",
    boldSuccessColor: "#1fcc44",
    successColor: "#4cd669",
    bgErrorColor: "#fae9e7",
    boldErrorColor: "#cc1f12",
    errorColor: "#d64c41",
    bgInfoColor: "#d0dbf5",
    boldInfoColor: "#124ccc",
    infoColor: "#5982db",
    bgWarningColor: "#fbf6e8",
    boldWarningColor: "#d9a318",
    warningColor: "#e1b546",
    bgNeutralColor: "#dbdee0",
    neutralColor: "#828b92",
    primaryDisableColor: "#6499c4"
}