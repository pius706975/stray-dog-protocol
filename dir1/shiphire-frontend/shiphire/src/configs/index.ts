export * from "./Fonts"
export * from "./Colors"
export * from "./LocalStorage"
export * from "./Svg"
export * from "./StatusText"
export * from "./firebase/GoogleSignIn"
export * from "./firebase/AppleSignIn"