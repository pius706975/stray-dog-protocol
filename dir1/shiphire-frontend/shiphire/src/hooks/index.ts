export * from './useAuth';
export * from './useRenter';
export * from './useShip';
export * from './useUser';
export * from './useHelper';
export * from './useShipOwner';
export * from './useAdmin';
export * from './useMutationWithErrorHandler';
