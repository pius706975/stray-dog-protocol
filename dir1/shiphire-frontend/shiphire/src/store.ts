import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
