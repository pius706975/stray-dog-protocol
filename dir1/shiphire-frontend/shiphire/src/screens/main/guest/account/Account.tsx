import React from 'react';
import { Platform } from 'react-native';
import { View } from 'react-native-ui-lib';
import { useTranslation } from 'react-i18next';
import { Button, ScreenLayout } from '../../../../components';
import { Color, LanguageIcon } from '../../../../configs';
import { GuestAccountProps } from '../../../../types';
import { AccountMenuItem } from './components';

const Account: React.FC<GuestAccountProps> = ({ navigation }) => {
    const plusPaddingV = Platform.OS === 'ios' ? 40 : 20;
    const { t } = useTranslation('account');

    return (
        <ScreenLayout
            backgroundColor="light"
            testId="guestHomeScreen"
            paddingV={plusPaddingV}>
            <View
                style={{
                    margin: 6,
                    borderRadius: 8,
                    backgroundColor: Color.bgNeutralColor,
                }}>
                <AccountMenuItem
                    onClick={() => navigation.navigate('ChangeLanguage')}
                    label={t('labelChangeLanguage')}
                    Icon={LanguageIcon}
                />
            </View>
            <View
                style={{
                    flexDirection: 'row',
                    alignSelf: 'center',
                }}>
                <View
                    style={{
                        marginTop: 350,
                        width: 150,
                    }}>
                    <Button
                        title={t('labelSignIn')}
                        onSubmit={() => {
                            navigation.navigate('SignIn');
                        }}
                    />
                </View>
            </View>
        </ScreenLayout>
    );
};

export default Account;
