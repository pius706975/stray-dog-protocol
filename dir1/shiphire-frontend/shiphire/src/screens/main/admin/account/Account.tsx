import React from 'react';
import { Button, ScreenLayout } from '../../../../components';
import { GoogleSignin } from '@react-native-google-signin/google-signin';
import { Platform } from 'react-native';
import {
    getDataFromLocalStorage,
    TOKEN,
    removeDataToLocalStorage,
    ROLES,
    Color,
    LanguageIcon,
    ADMINDATA,
} from '../../../../configs';
import { handleAxiosError } from '../../../../utils';
import { useSignOut, useUpdateFirebaseToken } from '../../../../hooks';
import { useDispatch } from 'react-redux';
import {
    modalSlice,
    progressIndicatorSlice,
    userStatusSlice,
} from '../../../../slices';
import { View } from 'react-native-ui-lib';
import { AccountMenuItem } from './components';
import { AccountProps } from '../../../../types';
import { useTranslation } from 'react-i18next';

const Account: React.FC<AccountProps> = ({ navigation }) => {
    const plusPaddingV = Platform.OS === 'ios' ? 40 : 20;
    const { t } = useTranslation('account');
    const mutationSignOut = useSignOut();
    const mutationUpdateFirebaseToken = useUpdateFirebaseToken();
    const dispatch = useDispatch();
    const { hideModal, showModal } = modalSlice.actions;
    const { hideProgressIndicator, showProgressIndicator } =
        progressIndicatorSlice.actions;
    const { logout } = userStatusSlice.actions;

    const updateAdminFirebaseToken = async () => {
        const payload = {
            token: '',
        };

        await mutationUpdateFirebaseToken.mutateAsync(payload, {
            onError: err => {
                handleAxiosError(err);
            },
        });
    };

    const handleLogOut = async () => {
        await getDataFromLocalStorage(TOKEN).then(resp => {
            mutationSignOut.mutate(
                {
                    refreshToken: resp.refreshToken,
                },
                {
                    onSuccess: async () => {
                        await updateAdminFirebaseToken();
                        removeDataToLocalStorage(TOKEN);
                        removeDataToLocalStorage(ROLES);
                        dispatch(
                            showModal({
                                status: 'success',
                                text: t('successSignOut'),
                            }),
                        );
                        dispatch(showProgressIndicator());
                        setTimeout(() => {
                            dispatch(hideModal());
                            dispatch(hideProgressIndicator());
                            dispatch(logout());
                        }, 2000);
                    },
                    onError: err => {
                        dispatch(
                            showModal({
                                status: 'failed',
                                text: t('failedSignOut'),
                            }),
                        );
                        setTimeout(() => {
                            dispatch(hideModal());
                        }, 2000);
                        handleAxiosError(err);
                    },
                },
            );
        });
    };
    return (
        <ScreenLayout
            backgroundColor="light"
            testId="adminHomeScreen"
            paddingV={plusPaddingV}>
            <View
                style={{
                    margin: 6,
                    borderRadius: 8,
                    backgroundColor: Color.bgNeutralColor,
                }}>
                <AccountMenuItem
                    onClick={() => navigation.navigate('ChangeLanguage')}
                    label={t('labelChangeLanguage')}
                    Icon={LanguageIcon}
                />
            </View>
            <View
                style={{
                    flexDirection: 'row',
                    alignSelf: 'center',
                }}>
                <View
                    style={{
                        marginTop: 350,
                        width: 150,
                    }}>
                    <Button title="Logout" onSubmit={handleLogOut} />
                </View>
            </View>
        </ScreenLayout>
    );
};

export default Account;
