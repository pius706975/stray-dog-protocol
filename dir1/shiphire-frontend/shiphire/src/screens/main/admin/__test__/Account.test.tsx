import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { QueryClient, QueryClientProvider } from 'react-query';
import {
    MainAdminStackParamList,
    MainScreenAdminParamList,
    MainScreenParamList,
} from '../../../../types';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import MockAdapter from 'axios-mock-adapter';
import httpRequest from '../../../../services/api';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import Account from '../account/Account';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { ADMINDATA, ROLES, TOKEN } from '../../../../configs';
import { useSignOut } from '../../../../hooks';
import renderer from 'react-test-renderer';

jest.mock('react-i18next', () => ({
    // this mock makes sure any components using the translate hook can use it without a warning being shown
    useTranslation: () => {
        return {
            t: str => str,
            i18n: {
                changeLanguage: () => new Promise(() => {}),
            },
        };
    },
    initReactI18next: {
        type: '3rdParty',
        init: () => {},
    },
}));
jest.mock('../../../../hooks');

const queryClient = new QueryClient();
const Tab = createBottomTabNavigator<MainScreenAdminParamList>();
const MainAdminStack = createNativeStackNavigator<MainAdminStackParamList>();
const mockAdapter = new MockAdapter(httpRequest);

const adminStackMockComponent = () => (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainAdminStack.Navigator>
                    <MainAdminStack.Screen name="AdminHome">
                        {() => (
                            <Tab.Navigator>
                                <Tab.Screen
                                    name="Account"
                                    component={Account}
                                />
                            </Tab.Navigator>
                        )}
                    </MainAdminStack.Screen>
                </MainAdminStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);
beforeEach(() => {
    AsyncStorage.setItem(
        ADMINDATA,
        JSON.stringify({
            name: 'Usmanul',
            email: 'usman@email.com',
            password: 'test',
            isVerified: false,
            isCompanySubmitted: false,
        }),
    );
    AsyncStorage.setItem(
        TOKEN,
        JSON.stringify({
            accessToken: 'test',
            refreshToken: 'test',
        }),
    );
    mockAdapter.reset();
});
describe('account screen', () => {
    describe('snapshot testing', () => {
        it('should render account screen correctly', () => {
            const accountScreen = render(adminStackMockComponent());
            expect(accountScreen).toMatchSnapshot();
        });
    });
    describe('unit testing', () => {
        it('should show modal error when failed logout', async () => {
            renderer.create(adminStackMockComponent());
            mockAdapter
                .onPost('/auth/signout')
                .reply(500, { status: 'failed' });
            (useSignOut as jest.Mock).mockImplementation(() => ({
                mutate: jest.fn().mockImplementation((data, options) => {
                    const { onError } = options;
                    const mockResponseData = {
                        message: 'failed',
                        status: 500,
                    };
                    onError(mockResponseData);
                }),
            }));
            const { findByText } = render(adminStackMockComponent());

            const logoutButton = await findByText('Logout');

            await waitFor(() => {
                fireEvent.press(logoutButton);
                const modalState = store.getState().modal;

                expect(modalState).toEqual({
                    visible: true,
                    status: 'failed',
                    text: 'failedSignOut',
                });
            });
        });
        it('should remove Token and Roles from AsyncStorage when logout success', async () => {
            mockAdapter
                .onPost('/auth/signout')
                .reply(200, { status: 'success' });
            (useSignOut as jest.Mock).mockImplementation(() => ({
                mutate: jest.fn().mockImplementation((data, options) => {
                    const { onSuccess } = options;
                    const mockResponseData = {
                        data: {
                            message: 'success',
                        },
                    };
                    onSuccess(mockResponseData);
                }),
            }));
            const { findByText } = render(adminStackMockComponent());

            const logoutButton = await findByText('Logout');

            fireEvent.press(logoutButton);

            await waitFor(() => {
                expect(AsyncStorage.removeItem).toHaveBeenCalledWith(TOKEN);
                expect(AsyncStorage.removeItem).toHaveBeenCalledWith(ROLES);

                const modalState = store.getState().modal;
                const userState = store.getState().userStatus.isLoggedIn;
                expect(userState).toBeFalsy();
                expect(modalState).toEqual({
                    visible: true,
                    status: 'success',
                    text: 'successSignOut',
                });
            });
        });
    });
});
