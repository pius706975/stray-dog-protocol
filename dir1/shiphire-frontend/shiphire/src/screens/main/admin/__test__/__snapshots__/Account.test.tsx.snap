// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account screen snapshot testing should render account screen correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <RNSScreenNavigationContainer
              style={
                {
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <RNSScreen
                activityState={2}
                collapsable={false}
                forwardedRef={[Function]}
                gestureResponseDistance={
                  {
                    "bottom": -1,
                    "end": -1,
                    "start": -1,
                    "top": -1,
                  }
                }
                onGestureCancel={[Function]}
                sheetAllowedDetents="large"
                sheetCornerRadius={-1}
                sheetExpandsWhenScrolledToEdge={true}
                sheetGrabberVisible={false}
                sheetLargestUndimmedDetent="all"
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "zIndex": 0,
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  importantForAccessibility="auto"
                  style={
                    [
                      {
                        "backgroundColor": "rgb(242, 242, 242)",
                        "flex": 1,
                      },
                      [
                        {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          {
                            "flex": 1,
                          },
                          undefined,
                          {},
                          {},
                          {},
                          {
                            "backgroundColor": "#f0f6fb",
                            "justifyContent": "space-between",
                            "position": "relative",
                          },
                        ]
                      }
                      testID="adminHomeScreen"
                    >
                      <RCTScrollView
                        contentContainerStyle={
                          {
                            "alignItems": undefined,
                            "flex": undefined,
                            "flexGrow": 1,
                            "gap": undefined,
                            "justifyContent": undefined,
                            "marginBottom": undefined,
                            "padding": undefined,
                            "paddingVertical": 40,
                          }
                        }
                      >
                        <View>
                          <View
                            style={
                              [
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                {
                                  "backgroundColor": "#dbdee0",
                                  "borderRadius": 8,
                                  "margin": 6,
                                },
                              ]
                            }
                          >
                            <View
                              accessibilityState={
                                {
                                  "busy": undefined,
                                  "checked": undefined,
                                  "disabled": undefined,
                                  "expanded": undefined,
                                  "selected": undefined,
                                }
                              }
                              accessibilityValue={
                                {
                                  "max": undefined,
                                  "min": undefined,
                                  "now": undefined,
                                  "text": undefined,
                                }
                              }
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                            >
                              <View
                                style={
                                  [
                                    undefined,
                                    undefined,
                                    undefined,
                                    undefined,
                                    undefined,
                                    undefined,
                                    undefined,
                                    {
                                      "alignItems": "center",
                                      "borderBottomWidth": 0.5,
                                      "flexDirection": "row",
                                      "justifyContent": "space-between",
                                      "margin": 5,
                                      "padding": 10,
                                      "paddingTop": 0,
                                    },
                                  ]
                                }
                              >
                                <View
                                  style={
                                    [
                                      undefined,
                                      undefined,
                                      undefined,
                                      undefined,
                                      undefined,
                                      undefined,
                                      undefined,
                                      {
                                        "flexDirection": "row",
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      [
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        undefined,
                                        {
                                          "justifyContent": "flex-start",
                                          "padding": 5,
                                          "paddingLeft": 0,
                                          "paddingRight": 10,
                                        },
                                      ]
                                    }
                                  >
                                    <RNSVGSvgView
                                      align="xMidYMid"
                                      bbHeight="24"
                                      bbWidth="24"
                                      focusable={false}
                                      height={24}
                                      meetOrSlice={0}
                                      minX={0}
                                      minY={-960}
                                      style={
                                        [
                                          {
                                            "backgroundColor": "transparent",
                                            "borderWidth": 0,
                                          },
                                          {
                                            "flex": 0,
                                            "height": 24,
                                            "width": 24,
                                          },
                                        ]
                                      }
                                      vbHeight={960}
                                      vbWidth={960}
                                      width={24}
                                      xml="<svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 -960 960 960" width="24"><path d="m476-80 182-480h84L924-80h-84l-42-122H604L560-80h-84Zm152-192h144l-70-198h-4l-70 198Zm-468 72-56-56 202-202q-38-42-66.5-87T190-640h84q18 36 38.5 65t49.5 61q44-48 73-98.5T484-720H40v-80h280v-80h80v80h280v80H564q-21 71-57 138t-89 126l96 98-30 82-124-124-200 200Z"/></svg>"
                                      xmlns="http://www.w3.org/2000/svg"
                                    >
                                      <RNSVGGroup
                                        fill={
                                          {
                                            "payload": 4278190080,
                                            "type": 0,
                                          }
                                        }
                                      >
                                        <RNSVGPath
                                          d="m476-80 182-480h84L924-80h-84l-42-122H604L560-80h-84Zm152-192h144l-70-198h-4l-70 198Zm-468 72-56-56 202-202q-38-42-66.5-87T190-640h84q18 36 38.5 65t49.5 61q44-48 73-98.5T484-720H40v-80h280v-80h80v80h280v80H564q-21 71-57 138t-89 126l96 98-30 82-124-124-200 200Z"
                                          fill={
                                            {
                                              "payload": 4278190080,
                                              "type": 0,
                                            }
                                          }
                                        />
                                      </RNSVGGroup>
                                    </RNSVGSvgView>
                                  </View>
                                  <Text
                                    numberOfLines={0}
                                    style={
                                      {
                                        "borderBottomWidth": 0,
                                        "borderColor": undefined,
                                        "color": "#4d5963",
                                        "fontFamily": "Kanit-Regular",
                                        "fontSize": 18,
                                        "lineHeight": undefined,
                                        "textAlign": undefined,
                                      }
                                    }
                                  >
                                    labelChangeLanguage
                                  </Text>
                                </View>
                                <RNSVGSvgView
                                  align="xMidYMid"
                                  bbHeight="24"
                                  bbWidth="24"
                                  fill="none"
                                  focusable={false}
                                  height={24}
                                  meetOrSlice={0}
                                  minX={0}
                                  minY={0}
                                  style={
                                    [
                                      {
                                        "backgroundColor": "transparent",
                                        "borderWidth": 0,
                                      },
                                      {
                                        "flex": 0,
                                        "height": 24,
                                        "width": 24,
                                      },
                                    ]
                                  }
                                  vbHeight={24}
                                  vbWidth={24}
                                  width={24}
                                  xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_77_30628)">
    <path d="M9.29 6.70978C8.9 7.09978 8.9 7.72978 9.29 8.11978L13.17 11.9998L9.29 15.8798C8.9 16.2698 8.9 16.8998 9.29 17.2898C9.68 17.6798 10.31 17.6798 10.7 17.2898L15.29 12.6998C15.68 12.3098 15.68 11.6798 15.29 11.2898L10.7 6.69978C10.32 6.31978 9.68 6.31978 9.29 6.70978Z" fill="#323232"/>
    </g>
    <defs>
    <clipPath id="clip0_77_30628">
    <rect width="24" height="24" fill="white"/>
    </clipPath>
    </defs>
    </svg>
             
    "
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <RNSVGGroup
                                    fill={null}
                                    propList={
                                      [
                                        "fill",
                                      ]
                                    }
                                  >
                                    <RNSVGGroup
                                      clipPath="clip0_77_30628"
                                      fill={
                                        {
                                          "payload": 4278190080,
                                          "type": 0,
                                        }
                                      }
                                    >
                                      <RNSVGPath
                                        d="M9.29 6.70978C8.9 7.09978 8.9 7.72978 9.29 8.11978L13.17 11.9998L9.29 15.8798C8.9 16.2698 8.9 16.8998 9.29 17.2898C9.68 17.6798 10.31 17.6798 10.7 17.2898L15.29 12.6998C15.68 12.3098 15.68 11.6798 15.29 11.2898L10.7 6.69978C10.32 6.31978 9.68 6.31978 9.29 6.70978Z"
                                        fill={
                                          {
                                            "payload": 4281479730,
                                            "type": 0,
                                          }
                                        }
                                        propList={
                                          [
                                            "fill",
                                          ]
                                        }
                                      />
                                    </RNSVGGroup>
                                    <RNSVGDefs>
                                      <RNSVGClipPath
                                        fill={
                                          {
                                            "payload": 4278190080,
                                            "type": 0,
                                          }
                                        }
                                        name="clip0_77_30628"
                                      >
                                        <RNSVGRect
                                          fill={
                                            {
                                              "payload": 4294967295,
                                              "type": 0,
                                            }
                                          }
                                          height="24"
                                          propList={
                                            [
                                              "fill",
                                            ]
                                          }
                                          width="24"
                                          x="0"
                                          y="0"
                                        />
                                      </RNSVGClipPath>
                                    </RNSVGDefs>
                                  </RNSVGGroup>
                                </RNSVGSvgView>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              [
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                undefined,
                                {
                                  "alignSelf": "center",
                                  "flexDirection": "row",
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                [
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  undefined,
                                  {
                                    "marginTop": 350,
                                    "width": 150,
                                  },
                                ]
                              }
                            >
                              <View
                                accessibilityState={
                                  {
                                    "busy": undefined,
                                    "checked": undefined,
                                    "disabled": undefined,
                                    "expanded": undefined,
                                    "selected": undefined,
                                  }
                                }
                                accessibilityValue={
                                  {
                                    "max": undefined,
                                    "min": undefined,
                                    "now": undefined,
                                    "text": undefined,
                                  }
                                }
                                accessible={true}
                                collapsable={false}
                                focusable={true}
                                onBlur={[Function]}
                                onClick={[Function]}
                                onFocus={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  {
                                    "alignItems": "center",
                                    "backgroundColor": "#216dab",
                                    "borderRadius": 6,
                                    "elevation": 2,
                                    "flexDirection": "row",
                                    "gap": 0,
                                    "justifyContent": "center",
                                    "paddingHorizontal": 10,
                                    "paddingVertical": 6,
                                  }
                                }
                              >
                                <Text
                                  numberOfLines={0}
                                  style={
                                    {
                                      "borderBottomWidth": 0,
                                      "borderColor": undefined,
                                      "color": "#f0f6fb",
                                      "fontFamily": "Kanit-Medium",
                                      "fontSize": 18,
                                      "lineHeight": undefined,
                                      "textAlign": undefined,
                                    }
                                  }
                                >
                                  Logout
                                </Text>
                                <View
                                  style={
                                    [
                                      undefined,
                                      undefined,
                                      undefined,
                                      undefined,
                                      undefined,
                                      undefined,
                                      undefined,
                                      {
                                        "position": "absolute",
                                        "right": 20,
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                          </View>
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      collapsable={false}
                      pointerEvents="box-none"
                      style={
                        {
                          "bottom": 0,
                          "left": 0,
                          "position": "absolute",
                          "right": 0,
                          "top": 0,
                          "zIndex": 0,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "backgroundColor": "rgb(255, 255, 255)",
                            "borderBottomColor": "rgb(216, 216, 216)",
                            "flex": 1,
                            "shadowColor": "rgb(216, 216, 216)",
                            "shadowOffset": {
                              "height": 0.5,
                              "width": 0,
                            },
                            "shadowOpacity": 0.85,
                            "shadowRadius": 0,
                          }
                        }
                      />
                    </View>
                    <View
                      collapsable={false}
                      pointerEvents="box-none"
                      style={
                        {
                          "height": 44,
                          "maxHeight": undefined,
                          "minHeight": undefined,
                          "opacity": undefined,
                          "transform": undefined,
                        }
                      }
                    >
                      <View
                        pointerEvents="none"
                        style={
                          {
                            "height": 0,
                          }
                        }
                      />
                      <View
                        pointerEvents="box-none"
                        style={
                          {
                            "alignItems": "stretch",
                            "flex": 1,
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          collapsable={false}
                          pointerEvents="box-none"
                          style={
                            {
                              "alignItems": "flex-start",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "justifyContent": "center",
                              "marginStart": 0,
                            }
                          }
                        />
                        <View
                          collapsable={false}
                          pointerEvents="box-none"
                          style={
                            {
                              "justifyContent": "center",
                              "marginHorizontal": 16,
                              "maxWidth": 718,
                            }
                          }
                        >
                          <Text
                            accessibilityRole="header"
                            aria-level="1"
                            collapsable={false}
                            numberOfLines={1}
                            style={
                              {
                                "color": "rgb(28, 28, 30)",
                                "fontSize": 17,
                                "fontWeight": "600",
                              }
                            }
                          >
                            Account
                          </Text>
                        </View>
                        <View
                          collapsable={false}
                          pointerEvents="box-none"
                          style={
                            {
                              "alignItems": "flex-end",
                              "flexBasis": 0,
                              "flexGrow": 1,
                              "justifyContent": "center",
                              "marginEnd": 0,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
              </RNSScreen>
            </RNSScreenNavigationContainer>
            <View
              collapsable={false}
              onLayout={[Function]}
              pointerEvents="auto"
              style={
                {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderTopColor": "rgb(216, 216, 216)",
                  "borderTopWidth": 0.5,
                  "bottom": 0,
                  "elevation": 8,
                  "height": 49,
                  "left": 0,
                  "paddingBottom": 0,
                  "paddingHorizontal": 0,
                  "position": null,
                  "right": 0,
                  "transform": [
                    {
                      "translateY": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <View
                accessibilityRole="tablist"
                style={
                  {
                    "flex": 1,
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityLabel="Account, tab, 1 of 1"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": true,
                    }
                  }
                  accessibilityStates={
                    [
                      "selected",
                    ]
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flex": 1,
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                      {
                        "flexDirection": "column",
                        "justifyContent": "flex-end",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      [
                        {
                          "flex": 1,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 1,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                            },
                            {
                              "color": "rgb(0, 122, 255)",
                              "fontSize": 25,
                            },
                            undefined,
                          ]
                        }
                      >
                        ⏷
                      </Text>
                    </View>
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "alignSelf": "center",
                            "height": "100%",
                            "justifyContent": "center",
                            "minWidth": 25,
                            "position": "absolute",
                            "width": "100%",
                          },
                          {
                            "opacity": 0,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "backgroundColor": "transparent",
                            },
                            {
                              "color": "#8E8E8F",
                              "fontSize": 25,
                            },
                            undefined,
                          ]
                        }
                      >
                        ⏷
                      </Text>
                    </View>
                  </View>
                  <Text
                    numberOfLines={1}
                    style={
                      [
                        {
                          "backgroundColor": "transparent",
                          "textAlign": "center",
                        },
                        {
                          "color": "rgb(0, 122, 255)",
                        },
                        {
                          "fontSize": 10,
                        },
                        undefined,
                      ]
                    }
                  >
                    Account
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="AdminHome"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
